### PR TITLE
feat: Aptos Da Client

### PR DESCRIPTION
**Pull request by Tranduy1dol**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ members = [
   "crates/da_clients/da-client-interface",
   "crates/da_clients/ethereum",
   "crates/utils",
-  "crates/settlement_clients/settlement-client-interface",
+  "crates/settlement_clients/settlement-client-interface", "crates/da_clients/aptos",
```
Comment (0x5ea000000) [inline Cargo.toml]: break line

File: Cargo.toml [hunk 2]
```diff
@@ -45,3 +46,8 @@ utils = { path = "crates/utils" }
 arc-swap = { version = "1.7.1" }
 num-traits = "0.2"
 lazy_static = "1.4.0"
+aptos-sdk = { git = "https://github.com/Tranduy1dol/aptos-core", branch="zeroize_v1.7" }
```
Comment (0x5ea000000) [inline Cargo.toml:50]: I will create an aptos core fork in sota-zk-labs later, add a todo to replace in the future

File: crates/da_clients/aptos/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+[package]
+name = "aptos-da-client"
+version.workspace = true
+edition.workspace = true
+authors.workspace = true
+
+[dependencies]
+color-eyre = "0.6.3"
```
Comment (0x5ea000000) [inline crates/da_clients/aptos/Cargo.toml]: follow the convention, add all dependencies to workspace

File: crates/da_clients/aptos/src/lib.rs
Comment (0x5ea000000) [inline crates/da_clients/aptos/src/lib.rs:1]: are all the contents of this file dummy and copypasta?

File: crates/da_clients/aptos/src/config.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            Url::from_str(
+                std::env::var("APTOS_NODE_URL")
+                    .as_ref()
+                    .map(|s| s.as_str())
+                    .unwrap_or("https://fullnode.devnet.aptoslabs.com"),
+            )
+                .unwrap()
+        });
+
+        static FAUCET_URL: Lazy<Url> = Lazy::new(|| {
```
Comment (0x5ea000000) [inline crates/da_clients/aptos/src/config.rs]: Faucet URL? don't bring unrelated part to our repo

Comment (Tranduy1dol) [reply crates/da_clients/aptos/src/config.rs]: yes sur

File: crates/da_clients/aptos/src/lib.rs
Comment (Tranduy1dol) [reply crates/da_clients/aptos/src/lib.rs:1]: no, this code is inspired by Ethereum DA layer.

File: Cargo.toml (hunk 2 above)
Comment (Tranduy1dol) [reply Cargo.toml:50]: I did that. Can you review it? Here is the [link](https://github.com/sota-zk-labs/aptos-core)

**Comment by 0x5ea000000** [comment]
@Tranduy1dol when you want to sync code from main branch, please use rebase instead of merge

File: Cargo.toml [hunk 5]
```diff
@@ -72,8 +74,14 @@ madara-prover-rpc-client = { git = "https://github.com/Moonsong-Labs/madara-prov
...
 utils = { path = "crates/utils" }
 prover-client-interface = { path = "crates/prover-services/prover-client-interface" }
 gps-fact-checker = { path = "crates/prover-services/gps-fact-checker" }
 sharp-service = { path = "crates/prover-services/sharp-service" }
 orchestrator = { path = "crates/orchestrator" }
+
+
+[patch.crates-io]
+merlin = { git = "https://github.com/aptos-labs/merlin" }
+x25519-dalek = { git = "https://github.com/Tranduy1dol/x25519-dalek", branch = "zeroize_v1.7" }
```
Comment (0x5ea000000) [inline Cargo.toml:54]: fork to our org

File: crates/da-clients/aptos/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "aptos-da-client"
+version.workspace = true
+edition.workspace = true
+
+[dependencies]
+alloy = { git = "https://github.com/alloy-rs/alloy", rev = "68952c0" }
+c-kzg = "1.0.2"
```
Comment (0x5ea000000) [inline crates/da-clients/aptos/Cargo.toml]: as the same as before, follow the convention, bring dependencies to workspace dependencies

**Review by 0x5ea000000 — COMMENTED** [review]
we didn't have the blob da contract part, maybe u can collaborate with @hduoc2003 to implement the avail and celestia DA, and open PR to main repo.

**Pull request closed without merging**